app/controllers/restaurants_controller.rb

```diff
@@ -40,4 +48,22 @@ def set_restaurant
     def restaurant_params
       params.require(:restaurant).permit(:name, :cuisine, :rating, :ten_bis, :address, :max_delivery_time)
     end
+
+  def create_where_params_obj(query_params)
```

Comment:
The naming convention is that a method name should be what it returns, without prefixes. (Also, `create` has a DB record connotation so I wouldn't use it when creating a simple object).
Also, I'de call it a `hash` and not an `obj`

Reply:
To make this method a bit more readable, and since that all it does is convert numbers to ranges when needed, I'd call it that.